lib/chef/knife/ssh.rb

```diff
@@ -279,7 +286,13 @@ def session_from_list(list)
       end
 
       def fixup_sudo(command)
-        command.sub(/^sudo/, 'sudo -p \'knife sudo password: \'')
+        if config[:ssh_password_ng] && config[:ssh_password_for_sudo]
```

Comment:
can you do `@password = config[:ssh_password_ng] if config[:ssh_password_for_sudo]`?

that will set `@password` and will avoid prompting, and then you'll fall through all the logic that we already have to find the `knife sudo password: ` string on the pty and send the password (which you definitely should not be bypassing -- its all there for $REASONS).

the code line to set `@password` probably should not be in fixup_sudo() and that command should be left alone.

not positive there won't be other side effects but you should go down that path rather than the one you're going down here.